extensions/Rules/lib/Rule.pm

```diff
@@ -0,0 +1,212 @@
+# This Source Code Form is hasject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::Rules::Rule;
+
+use 5.10.1;
+use Data::Dumper;
+use Moo;
+use List::Util qw(any none);
+use Types::Standard -all;
+use Type::Utils;
+
+use Bugzilla::Logging;
+use Bugzilla::Status;
+use Bugzilla::Types qw(Bug FakeBug User);
+
+#########################
+#    Initialization     #
+#########################
+
+# Rule definitions
+has action    => (is => 'ro', isa => ArrayRef | Str);
+has change    => (is => 'ro', isa => Maybe [Dict]);
+has condition => (is => 'ro', isa => Maybe [Dict]);
+has error     => (is => 'ro', isa => Maybe [Str]);
+has filter    => (is => 'ro', isa => Maybe [Dict]);
+has name      => (is => 'ro', isa => Str);
+
+# Current attributes
+has bug       => (is => 'ro', isa => Bug | FakeBug);
+has user      => (is => 'ro', isa => User);
+has field     => (is => 'ro', isa => Str);
+has old_value => (is => 'ro', isa => Str);
+has new_value => (is => 'ro', isa => Str);
+
+sub BUILDARGS {
+  my ($class, $params) = @_;
+
+  $params->{action}    = $params->{rule}->{action};
+  $params->{change}    = $params->{rule}->{change};
+  $params->{condition} = $params->{rule}->{condition};
+  $params->{error}     = $params->{rule}->{error};
+  $params->{filter}    = $params->{rule}->{filter};
+  $params->{name}      = $params->{rule}->{name};
```

Comment:
If the TOML library doesn't `trim` whitespace, we should do that here for all of the fields.